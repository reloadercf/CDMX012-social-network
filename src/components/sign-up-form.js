// eslint-disable-next-line import/no-cycle
import { createAccount } from '../lib/auth.js';
import { showIncorrectPass, showPassword } from './ui.js';

export const createAccForm = () => {
  const createAccContainer = document.createElement('div');
  createAccContainer.setAttribute('class', 'createAccount-container');

  const createAccTitle = document.createElement('h2');
  createAccTitle.setAttribute('class', 'account-title');
  createAccTitle.innerHTML = 'Create an Account';

  const signUpForm = document.createElement('form');
  signUpForm.setAttribute('class', 'account-form');

  const emailFormat = document.createElement('div');
  emailFormat.setAttribute('class', 'label-icon');
  emailFormat.innerHTML = `
    <img class="mail" src='./assets/icons8-email-64 1.png' alt="mail"/>
    <label for="email" class="e-mail">E-mail</label>`;

  const txtEmail = document.createElement('input');
  txtEmail.setAttribute('name', 'email');
  txtEmail.setAttribute('class', 'e-mail');
  txtEmail.setAttribute('type', 'email');
  txtEmail.setAttribute('placeholder', 'e-mail@example.com');
  txtEmail.required = 'true';
  txtEmail.id = 'txtEmail';

  const passwordFormat = document.createElement('div');
  passwordFormat.setAttribute('class', 'label-icon');
  passwordFormat.innerHTML = `
    <img class="padlock" src='./assets/icons8-contraseña-24 1.png' alt="padlock"/>
    <label for="password" class="password" title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters">Password</label>`;

  const eyeIcon1 = document.createElement('span');
  eyeIcon1.setAttribute('class', 'material-icons-outlined eye-icon');
  eyeIcon1.setAttribute('id', 'eyeIcon1');
  eyeIcon1.innerHTML = 'visibility';
  passwordFormat.append(eyeIcon1);

  const txtPassword = document.createElement('input');
  txtPassword.setAttribute('name', 'password');
  txtPassword.setAttribute('class', 'password');
  txtPassword.setAttribute('type', 'password');
  txtPassword.setAttribute('placeholder', 'Insert your password here');
  txtPassword.required = 'true';
  txtPassword.id = 'txtPassword';

  const passConfForm = document.createElement('div');
  passConfForm.setAttribute('class', 'label-icon');
  passConfForm.innerHTML = `
    <img class="padlock" src='./assets/icons8-contraseña-24 1.png' alt="padlock"/>
    <label for="passwordConfirm" class="password" title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters">Confirm your password</label>`;

  const eyeIcon2 = document.createElement('span');
  eyeIcon2.setAttribute('class', 'material-icons-outlined eye-icon');
  eyeIcon2.setAttribute('id', 'eyeIcon2');
  eyeIcon2.innerHTML = 'visibility';
  passConfForm.append(eyeIcon2);

  const passwordConf = document.createElement('input');
  passwordConf.setAttribute('name', 'passwordConfirm');
  passwordConf.setAttribute('class', 'password');
  passwordConf.setAttribute('type', 'password');
  passwordConf.setAttribute('placeholder', 'Confirm your password');
  passwordConf.required = 'true';
  passwordConf.id = 'passwordConfirm';

  signUpForm.append(emailFormat, txtEmail, passwordFormat, txtPassword, passConfForm, passwordConf);

  const errorSection = document.createElement('p');
  errorSection.setAttribute('class', 'error show-error-msg');
  errorSection.setAttribute('id', 'errorArea');

  const btnSignUp = document.createElement('input');
  btnSignUp.setAttribute('type', 'button');
  btnSignUp.setAttribute('class', 'btn-signUp');
  btnSignUp.setAttribute('value', 'Sign Up');

  createAccContainer.append(createAccTitle, signUpForm, errorSection, btnSignUp);

  // Password verification
  txtPassword.addEventListener('keyup', () => showIncorrectPass(txtPassword.value, passwordConf.value));
  passwordConf.addEventListener('keyup', () => showIncorrectPass(txtPassword.value, passwordConf.value));

  // Mask and unmask passwords
  eyeIcon1.addEventListener('click', () => showPassword(txtPassword, eyeIcon1));
  eyeIcon2.addEventListener('click', () => showPassword(passwordConf, eyeIcon2));

  // Sign up with email
  btnSignUp.addEventListener('click', () => {
    const email = document.getElementById('txtEmail').value;
    const password = document.getElementById('txtPassword').value;

    createAccount(email, password);
  });

  return createAccContainer;
};
