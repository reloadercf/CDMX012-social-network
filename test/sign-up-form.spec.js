/**
 * @jest-environment jsdom
 */
import { createAccForm } from '../src/components/sign-up-form.js';

jest.mock('../src/firebase-imports.js');

describe('Tests of account form', () => {
  test('render create account form inside sign-up page', () => {
    const renderCreateAccForm = createAccForm();
    document.body.innerHTML = '<div class="lp-content"></div>';
    const lpContent = document.querySelector('.lp-content');
    lpContent.appendChild(renderCreateAccForm);
    expect(lpContent.innerHTML).toMatchSnapshot();
  });
});
