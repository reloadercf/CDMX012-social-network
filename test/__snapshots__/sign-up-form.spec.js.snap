// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of account form render create account form inside sign-up page 1`] = `
"<div class=\\"createAccount-container\\"><h2 class=\\"account-title\\">Create an Account</h2><form class=\\"account-form\\"><div class=\\"label-icon\\">
    <img class=\\"mail\\" src=\\"./assets/icons8-email-64 1.png\\" alt=\\"mail\\">
    <label for=\\"email\\" class=\\"e-mail\\">E-mail</label></div><input name=\\"email\\" class=\\"e-mail\\" type=\\"email\\" placeholder=\\"e-mail@example.com\\" required=\\"\\" id=\\"txtEmail\\"><div class=\\"label-icon\\">
    <img class=\\"padlock\\" src=\\"./assets/icons8-contraseña-24 1.png\\" alt=\\"padlock\\">
    <label for=\\"password\\" class=\\"password\\" title=\\"Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters\\">Password</label><span class=\\"material-icons-outlined eye-icon\\" id=\\"eyeIcon1\\">visibility</span></div><input name=\\"password\\" class=\\"password\\" type=\\"password\\" placeholder=\\"Insert your password here\\" required=\\"\\" id=\\"txtPassword\\"><div class=\\"label-icon\\">
    <img class=\\"padlock\\" src=\\"./assets/icons8-contraseña-24 1.png\\" alt=\\"padlock\\">
    <label for=\\"passwordConfirm\\" class=\\"password\\" title=\\"Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters\\">Confirm your password</label><span class=\\"material-icons-outlined eye-icon\\" id=\\"eyeIcon2\\">visibility</span></div><input name=\\"passwordConfirm\\" class=\\"password\\" type=\\"password\\" placeholder=\\"Confirm your password\\" required=\\"\\" id=\\"passwordConfirm\\"></form><p class=\\"error show-error-msg\\" id=\\"errorArea\\"></p><input type=\\"button\\" class=\\"btn-signUp\\" value=\\"Sign Up\\"></div>"
`;
